pkg/api/converterfromapi.go

```diff
@@ -578,3 +613,30 @@ func convertCertificateProfileToVLabs(api *CertificateProfile, vlabs *vlabs.Cert
 	vlabs.KubeConfigPrivateKey = api.KubeConfigPrivateKey
 	vlabs.SetCAPrivateKey(api.GetCAPrivateKey())
 }
+
+func addDCOSPublicAgentPool(privateAgentPool *AgentPoolProfile, p *Properties, m *MasterProfile) (*AgentPoolProfile, error) {
+	privateAgentPool.DNSPrefix = "" // - don't use DNS for private pool
+	// TODO - VM Count for private pool matches api model
+	// p.AgentPoolProfiles[0].Count = ??? <-- TODO
+	publicPool := &AgentPoolProfile{}
+	publicPool.Name = privateAgentPool.Name + "_public"
+	publicPool.VMSize = privateAgentPool.VMSize // - use same VMsize for public pool
+	publicPool.OSType = privateAgentPool.OSType
+	for _, port := range [3]int{80, 443, 8080} {
+		fmt.Println(port)
+		publicPool.Ports = append(publicPool.Ports, port)
+	}
+	fmt.Println("%#v\n", publicPool.Ports)
+	// - VM Count for public agents is based on the following:
+	// 1 master => 1 VM
+	// 3, 5 master => 3 VMsize
+	if m.Count == 1 {
+		publicPool.Count = 1
+	} else if m.Count == 3 {
+		publicPool.Count = 3
+	} else if m.Count == 5 {
+		publicPool.Count = 3
+	}
+	// TODO how do we set v20160930PublicPool.Count if p.MasterProfile.Count is neither 1, 3, nor 5?
```

Comment:
we have validation that only 1, 3, or 5 is allowed. With no plan of changing at this time.